GeoLib/QuadTree.h

```diff
@@ -381,16 +384,20 @@ template <typename POINT> class QuadTree
 		POINT mid_point(_ll);
 		mid_point[0] += (_ur[0] - _ll[0]) / 2.0;
 		mid_point[1] += (_ur[1] - _ll[1]) / 2.0;
+		assert(_childs[0] == nullptr);
```

Comment:
My arguments would be that they are interupting the flow when reading the code and that code is executed even though you could already have known that something is wrong. If something is actually wrong you need to debug anyway. But it's no big issue, do it however you like.